systemverilog-plugin/UhdmAst.cc

```diff
@@ -3587,8 +3587,12 @@ void UhdmAst::process_assignment_pattern_op()
         visit_one_to_many({vpiOperand}, obj_h, [&](AST::AstNode *node) {
             if (node->type == AST::AST_ASSIGN || node->type == AST::AST_ASSIGN_EQ || node->type == AST::AST_ASSIGN_LE) {
                 // Find at what position in the concat should we place this node
-                auto key = node->children[0]->str;
-                key = key.substr(key.find('.') + 1);
+                std::string key;
+                if (node->children[0]->children[0]->type == static_cast<AST::AstNodeType>(AST::Extended::AST_DOT)) {
+                    key = node->children[0]->children[0]->str;
+                } else {
+                    key = node->children[0]->str;
+                }
```

Comment:
Please add guards to make sure correct nodes exists, e.g.:
```suggestion
                if (!node->children.empty() && !node->children[0]->children.empty() && node->children[0]->children[0]->type == static_cast<AST::AstNodeType>(AST::Extended::AST_DOT)) {
                    key = node->children[0]->children[0]->str;
                } else if (!node->children.empty()) {
                    key = node->children[0]->str;
                } else {
                    log_file_error(node->filename, node->location.first_line,
                               "Couldn't find `key` in assignment pattern.\n");
                }
```

Reply:
Done.